modepy/matrices.py

```diff
@@ -351,6 +354,77 @@ def mass_matrix(
     return la.inv(inverse_mass_matrix(basis, nodes))
 
 
+def modal_quad_bilinear_form(
```

Comment:
I'm thinking about the naming of this. I'm not sure "bilinear form" is that great a fit, because the trial functions don't appear. I thought of "projection", but that also misses the mark, because the test functions could be derivatives. In a word, I'm not sure. I can probably be convinced to go along with "bilinear form", but I figured it might be good to at least discuss the naming.

(@alexfikl, if you're reading along, please feel free to chime in too)

Reply:
I'm also not the biggest fan of the name, but I couldn't come up with anything better. `modal_quadrature_operator` maybe?

Reply:
Where is this `modal_quad_bilinear_form` supposed to be used? I'm guessing the `nodal_quad_bilinear_form` is used to define a proper bilinear form somewhere?

Reply:
`modal_quad_bilinear_form` doesn't have any uses right now other than returning the operator that's transformed into a nodal operator.

`nodal_quad_bilinear_form` will be used in a grudge update that evaluates operators using quadrature (https://github.com/inducer/grudge/pull/362 is the staging area for this, although a little outdated at this point). While `nodal_quadrature_operator` could be used for everything, `nodal_quad_bilinear_form` makes the corresponding grudge code cleaner when there is no overintegration.

TLDR; `modal_quad_bilinear_form`'s current use is in `nodal_quad_bilinear_form`. `nodal_quad_bilinear_form` is used to evaluate DG operators in grudge when there is no overintegration. `nodal_quadrature_operator` is used when there is overintegration.

Also, the original comment is outdated now since `modal_quad_bilinear_form` *does* use trial functions (as of [c063d23](https://github.com/inducer/modepy/pull/103/commits/c063d234325d04823773c3ea614ee7f2e0bb8586)).